[FLINK-17704][mvn] allow execution of arbitrary benchmarks via a property

NicoK opened the pull request:
This is based on #59 and adds a simple way of executing an arbitrary subset of benchmarks, e.g.

```
mvn -Dflink.version=1.11-SNAPSHOT exec:exec -Dbenchmarks="org.apache.flink.benchmark.StreamNetworkThroughputBenchmarkExecutor"
```

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dataartisans/flink-benchmarks/60)
<!-- Reviewable:end -->

NicoK committed:
[FLINK-17704][mvn] allow execution of arbitrary benchmarks via a property

Example:
mvn -Dflink.version=1.11-SNAPSHOT exec:exec -Dbenchmarks="org.apache.flink.benchmark.StreamNetworkThroughputBenchmarkExecutor"

pnowojski commented:
Wasn't there already a way to do it? It was written down in the readme:
https://github.com/dataArtisans/flink-benchmarks
> mvn -Dflink.version=1.5.0 clean package exec:exec \
 -Dexec.executable=java -Dexec.args="-jar target/benchmarks.jar -rf csv org.apache.flink.state.benchmark.*"

At least the README should be updated.

NicoK committed:
[hotfix] update flink version in README

NicoK commented:
yes, the previously mentioned way was basically the same as executing "java -jar ..." yourself with all the knowledge required to run it (except for the classpath being added automatically). The new way should be much simpler and less error-prone in case of changes.

I updated the README and also changed the implementation to allow benchmarks that are excluded in the default profile.

To verify the profiles are working, I ran these commands and compared the resulting command lines, active profiles, etc:

```
mvn -Dflink.version=1.11-SNAPSHOT exec:exec clean package -Dbenchmarks="org.apache.flink.state.benchmark.ValueStateBenchmark.valueUpdate" -e -X
mvn -Dflink.version=1.11-SNAPSHOT exec:exec clean package -e -X
mvn help:active-profiles -P generate-thrift
mvn clean package -P generate-thrift
```
